socketio/src/client/client.rs

```diff
@@ -468,6 +568,53 @@ mod test {
         Ok(())
     }
 
+    #[test]
+    fn socket_io_reconnect_integration() -> Result<()> {
+        let url = crate::test::socket_io_restart_server();
+
+        let socket = ClientBuilder::new(url).connect();
+        assert!(socket.is_ok(), "should connect success");
+        let socket = socket.unwrap();
+
+        std::thread::sleep(std::time::Duration::from_millis(200));
+
+        let r = socket.emit("message", json!(""));
+        assert!(r.is_ok(), "should emit message success");
+
+        let r = socket.emit("restart_server", json!(""));
+        assert!(r.is_ok(), "should emit restart success");
+
+        let mut is_emit_error = false;
+        for _ in 0..10 {
+            std::thread::sleep(std::time::Duration::from_millis(500));
+            let r = socket.emit("message", json!(""));
+            if r.is_err() {
+                is_emit_error = true; // server is down
+                break;
+            }
```

Comment:
(usability question) nit: should the re-connection be completely transparent to end user? As a user if I had to attempt to repeat all of my emits that may make it hard to utilize this feature